tests/unit/cmor/test_cmor_check.py

```diff
@@ -918,6 +921,78 @@ def test_frequency_not_supported(self):
         """Fail at metadata if frequency is not supported."""
         self._check_fails_in_metadata(frequency='wrong_freq')
 
+    def test_hourly_time_bounds(self):
+        """Test time bounds are properly guessed for hourly freq"""
+        cube = self.get_cube(self.var_info, frequency='hr')
+        original_bounds = cube.coord('time').bounds
+        cube.coord('time').bounds = None
+        self.cube = cube
+        self._check_cube(automatic_fixes=True, frequency='hr')
+        guessed_bounds = self.cube.coord('time').bounds
+        assert original_bounds.all() == guessed_bounds.all()
+
+    def test_3hourly_time_bounds(self):
+        """Test time bounds are properly guessed for 3hourly freq"""
+        cube = self.get_cube(self.var_info, frequency='3hr')
+        original_bounds = self.cube.coord('time').bounds
+        cube.coord('time').bounds = None
+        self.cube = cube
+        self._check_cube(automatic_fixes=True, frequency='3hr')
+        guessed_bounds = self.cube.coord('time').bounds
+        assert original_bounds.all() == guessed_bounds.all()
+
+    def test_6hourly_time_bounds(self):
+        """Test time bounds are properly guessed for 6hourly freq"""
+        cube = self.get_cube(self.var_info, frequency='6hr')
+        original_bounds = self.cube.coord('time').bounds
+        cube.coord('time').bounds = None
+        self.cube = cube
+        self._check_cube(automatic_fixes=True, frequency='6hr')
+        guessed_bounds = self.cube.coord('time').bounds
+        assert original_bounds.all() == guessed_bounds.all()
+
+    def test_daily_time_bounds(self):
+        """Test time bounds are properly guessed for daily freq"""
+        original_bounds = self.cube.coord('time').bounds
+        self.cube.coord('time').bounds = None
+        self._check_cube(automatic_fixes=True)
+        guessed_bounds = self.cube.coord('time').bounds
+        assert original_bounds.all() == guessed_bounds.all()
+
+    def test_monthly_time_bounds(self):
+        """Test time bounds are guessed for monthly freq"""
+        cube = self.get_cube(self.var_info, frequency='mon')
+        cube.coord('time').bounds = None
+        self.cube = cube
+        self._check_cube(automatic_fixes=True, frequency='mon')
+        assert self.cube.coord('time').bounds is not None
+
+    def test_yearly_time_bounds(self):
+        """Test time bounds are guessed for yearly freq"""
+        cube = self.get_cube(self.var_info, frequency='yr')
```

Comment:
Maybe you could just loop over the frequencies and condense this to a single test?

Reply:
Sure.